src/test/mir-opt/const_allocation3/rustc.main.ConstProp.after.mir

```diff
@@ -0,0 +1,51 @@
+// MIR for `main`
+// source = MirSource { instance: Item(DefId(0:3 ~ const_allocation3[317d]::main[0])), promoted: None }
+// pass_name = ConstProp
+// disambiguator = after
+
+fn  main() -> () {
+    let mut _0: ();                      // return place in scope 0 at $DIR/const_allocation3.rs:4:11: 4:11
+    let _1: &Packed;                     // in scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+    let mut _2: &&Packed;                // in scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+
+    bb0: {
+        StorageLive(_1);                 // bb0[0]: scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+        StorageLive(_2);                 // bb0[1]: scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+        _2 = const Scalar(alloc2+0): &&Packed; // bb0[2]: scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+                                         // ty::Const
+                                         // + ty: &&Packed
+                                         // + val: Value(Scalar(alloc2+0))
+                                         // mir::Constant
+                                         // + span: $DIR/const_allocation3.rs:5:5: 5:8
+                                         // + literal: Const { ty: &&Packed, val: Value(Scalar(alloc2+0)) }
+        _1 = (*_2);                      // bb0[3]: scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+        StorageDead(_2);                 // bb0[4]: scope 0 at $DIR/const_allocation3.rs:5:8: 5:9
+        StorageDead(_1);                 // bb0[5]: scope 0 at $DIR/const_allocation3.rs:5:8: 5:9
+        _0 = ();                         // bb0[6]: scope 0 at $DIR/const_allocation3.rs:4:11: 6:2
+        return;                          // bb0[7]: scope 0 at $DIR/const_allocation3.rs:6:2: 6:2
+    }
+}
+
+alloc2 (static: FOO, size: 8, align: 8) {
+    ├──────alloc9+0───────┤
+}
+
+alloc9 (size: 150, align: 1) {
+    0x00 │ ab ab ab ab ab ab ab ab ab ab ab ab ab ab ab ab
+    0x10 │ ab ab ab ab ab ab ab ab ab ab ab ab ├─alloc6+0─
+    0x20 │ ──────────┤ 01 ef cd ab 00 00 00 00 00 00 00 00
+    0x30 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x40 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x50 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x60 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x70 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x80 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ├alloc8+0 (8 bytes)
```

Comment:
Ah... I thought this should say "2 bytes" because there are only 2 bytes on this line. But the entire pointer still has 8 bytes. So I am not actually sure now.

Reply:
I think the `(N bytes)` thing is just confusing (to me it sounds like it talks about the allocation size). I'm just adding it because you want it ;). So whatever you want me to do I'll do it.

Reply:
> to me it sounds like it talks about the allocation size

Hm, I hadn't thought about that.
It's not like I know the right answer here, I am just trying to come up with something reasonable.^^ I'm open to better suggestions. :)

Reply:
The only one I had was the `(0)` `(1)` indexing one that shows the detailed info in a comment at the end of the row or in the next row.

Alternatively we can use something similar to the previous non-inline printing

Reply:
Print in a shell to have it look pretty :D

```
alloc9 (size: 150, align: 1) {
    0x00 │ ab ab ab ab ab ab ab ab ab ab ab ab ab ab ab ab
    0x10 │ ab ab ab ab ab ab ab ab ab ab ab ab ├─alloc6+0─
    0x20 │ ──────────┤ 01 ef cd ab 00 00 00 00 00 00 00 00
    0x30 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
    0x40 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
    0x50 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
    0x60 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
    0x70 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
    0x80 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ├──┬─
                                      ┌─────────────────┴┐
                                      │alloc8+0 (8 bytes)│
                                      └┬─────────────────┘
                   ┌───────────────────┘
    0x90 │ ────────┴───────┤
}
```

Reply:
oO